syft/formats/common/spdxhelpers/to_format_model.go

```diff
@@ -123,13 +148,113 @@ func ToFormatModel(s sbom.SBOM) *spdx.Document {
 			// Cardinality: optional, one
 			CreatorComment: "",
 		},
-		Packages:      toPackages(s.Artifacts.Packages, s),
+		Packages:      packages,
 		Files:         toFiles(s),
 		Relationships: relationships,
 		OtherLicenses: toOtherLicenses(s.Artifacts.Packages),
 	}
 }
 
+func toRootRelationships(rootPackage *spdx.Package, packages []*spdx.Package) (out []*spdx.Relationship) {
+	for _, p := range packages {
+		out = append(out, &spdx.Relationship{
+			RefA: spdx.DocElementID{
+				ElementRefID: rootPackage.PackageSPDXIdentifier,
+			},
+			Relationship: string(ContainsRelationship),
+			RefB: spdx.DocElementID{
+				ElementRefID: p.PackageSPDXIdentifier,
+			},
+		})
+	}
+	return
+}
+
+//nolint:funlen
+func toRootPackage(s sbom.SBOM) *spdx.Package {
+	var prefix string
+
+	name := s.Source.Name
+	nameIfUnset := func(n string) {
+		if name != "" {
+			return
+		}
+		name = n
+	}
+
+	version := s.Source.Version
+	versionIfUnset := func(v string) {
+		if version != "" && version != "latest" {
+			return
+		}
+		version = v
+	}
+
+	purpose := ""
+	var checksums []spdx.Checksum
+	switch m := s.Source.Metadata.(type) {
+	case source.StereoscopeImageSourceMetadata:
+		prefix = prefixImage
+		ref, err := reference.Parse(m.UserInput)
+		if err != nil {
+			log.Debugf("unable to parse image ref: %s", m.UserInput)
+			break
+		}
+		if ref, ok := ref.(reference.Named); ok {
+			nameIfUnset(ref.Name())
+		} else {
+			nameIfUnset(m.UserInput)
+		}
+		if ref, ok := ref.(reference.NamedTagged); ok {
+			versionIfUnset(ref.Tag())
+		}
+		if ref, ok := ref.(reference.Digested); ok {
+			versionIfUnset(ref.Digest().String())
+		}
+		versionIfUnset(m.ID)
+		purpose = spdxPrimaryPurposeContainer
+
+		c := toChecksum(m.ID)
```

Comment:
should this be the digest relative to the container image manifest and not the ID? Though the ID is a checksum, it isn't semantic to the artifact, but instead derived within syft